src/RHI/RHICommandList.h

```diff
@@ -54,6 +56,10 @@ class RHICommandListInterface
     virtual void Transition(std::span<std::pair<RHITexture&, RHITextureState::Flags>> textureNewStatePairs) = 0;
     virtual void Transition(std::span<std::pair<RHIBuffer&, RHIBufferState::Flags>> bufferNewStatePairs) = 0;
 
+    // Need to be called before and after all draw commands with the same DrawBinding
+    virtual void BeginRendering(const RHIDrawResources& resources) = 0;
+    virtual void EndRendering() = 0;
```

Comment:
This is probably a great idea, we could potentially change the entire CommandContext draw/dispatch structure to reproduce something similar to this? Would avoid having to cache PSOs and various other states etc...
It makes for a more optimized (and lean) abstraction as well.